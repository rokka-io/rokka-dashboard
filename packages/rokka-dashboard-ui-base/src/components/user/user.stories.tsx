import { storiesOf } from '@storybook/react';
import React from 'react';
import { colors } from '../../identity/colors/colors';
import { User } from './user';

storiesOf('Components / Navigation', module)
  .addDecorator(story => (
    <div style={{ width: '100vw', height: '50px', padding: '10px', background: colors.brand.primary }}>{story()}</div>
  ))
  .add('User', () => <User username="rokka-dashboard" />);
