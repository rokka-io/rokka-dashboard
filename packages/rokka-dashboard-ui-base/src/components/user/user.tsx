import React from 'react';
import styled from 'styled-components';
import { AvatarIcon, SpanWhite } from '../../elements';
import { colors } from '../../identity/colors/colors';
import { spaces } from '../../identity/spaces/spaces';

interface UserProps {
  /** Username */
  username: string;
}

export function User({ username }: UserProps) {
  return (
    <StyledUserDiv>
      <AvatarDiv>
        <AvatarIcon width="32px" height="32px" color={colors.brand.avatar} />
      </AvatarDiv>
      <StyledUsername>{username}</StyledUsername>
    </StyledUserDiv>
  );
}

const StyledUserDiv = styled.div`
  line-height: inherit;
  float: left;
`;

const StyledUsername = styled(SpanWhite)`
  margin-left: ${spaces.small};
`;

const AvatarDiv = styled.div`
  border-radius: 50%;
  background: ${colors.tints.white};
  box-shadow: 0 0 0 3px rgba(0, 0, 0, 0.2);
  overflow: hidden;
  display: inline-block;
  height: ${spaces.large};
  line-height: ${spaces.large};
  vertical-align: middle;
`;
