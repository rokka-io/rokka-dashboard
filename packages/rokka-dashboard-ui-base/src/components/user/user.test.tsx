import { shallow } from 'enzyme';
import React from 'react';

import { User } from './user';

describe('Components', () => {
  describe('User', () => {
    it('renders a user correctly.', () => {
      const component = shallow(<User username="rokka-test" />);

      expect(component).toMatchSnapshot();
    });
  });
});
