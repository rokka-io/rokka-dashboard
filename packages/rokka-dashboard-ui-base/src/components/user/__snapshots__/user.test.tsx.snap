// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components User renders a user correctly. 1`] = `
<styled.div>
  <styled.div>
    <AvatarIcon
      color="#588536"
      height="32px"
      width="32px"
    />
  </styled.div>
  <Styled(styled.span)>
    rokka-test
  </Styled(styled.span)>
</styled.div>
`;
